src/tools.rs

```diff
@@ -237,12 +237,16 @@ impl AppCache {
 
 /// Locate the given application and download it if missing.
 #[tracing::instrument(level = "trace")]
-pub async fn get(app: Application, version: Option<&str>) -> Result<PathBuf> {
+pub async fn get(app: Application, version: Option<&str>, offline: bool) -> Result<PathBuf> {
     if let Some((path, version)) = find_system(app, version).await {
         tracing::info!(app = %app.name(), %version, "using system installed binary");
         return Ok(path);
     }
 
+    if offline {
+        return Err(anyhow!("couldn't find application {}", &app.name()));
```

Comment:
This error string needs to be clarified a bit:
```suggestion
        return Err(anyhow!("couldn't find application {} and offline mode is enabled", &app.name()));
```